UsingTheServer.md

```diff
@@ -22,54 +22,170 @@ repositories {
+
+Below is a simple example of how server setup might look:
+
+```java
+new HttpServer(logger)
+     .addStatic(root)
+     .router(new Routes()
+                  .get("/redirect", new RedirectController())
+                  .get("/form", new FormController())
+                  .post("/form", new FormController())
+                  .put("/form", new FormController())
+                  .delete("/form", new FormController())
+                  .get("/logs", new LogController()))
+     .use(new AuthMiddleware("admin", "hunter2", "ServerCity", authRoutes))
+     .start(port);
+
+```
+
+Notes: Controllers are application specific.  `authRoutes` is a list of routes which require authorization and is application specific.
+
+## Setting Up Routes
+
+To set up routes for the server, create an instance of Routes: `new Routes()`.
+
+Specific routes can be added to the routes instance using the following methods:
+
+`.get(String uri, Controller controller)`    
+`.post(String uri, Controller controller)`    
+`.put(String uri, Controller controller)`    
+`.head(String uri, Controller controller)`    
+`.options(String uri, Controller controller)`    
+`.delete(String uri, Controller controller)`    
+
+Methods correspond to their respective HTTP method: GET, POST, PUT, HEAD, OPTIONS, DELETE.
+
+### Example Routes
+
+As an example, to create a routes instance which handles a get request to "/about" and a post request to "/form", we would configure our routes as follows:
+
+```java
+new Routes()
+     .get("/about", aboutController)
+     .post("/form", formController)
+```
+
+##Setting Up Controllers
+
+Controllers must implement the `Controller` interface.
+
+### Example Controller
+
+An example `Controller` from the cobspec application is provided below:
+
+```java
+public class RedirectController implements Controller {
+    public Response createResponse(Request request) {
+        return new ResponseBuilder()
+                    .addStatus(302)
+                    .addHeader("Location", "/")
+                    .build();
+    }
+}
+```
+
+Use of the `ResponseBuilder` is not required but is provided for convenience.
+
+Your `Controller` may contain whatever additional methods are needed to create contents of your response.
+
+## Setting Up Middleware
+
+Custom middleware must implement the `Middleware` interface.  `use` should return a function which takes a request and either returns a custom response or calls `createResponse` for the `controller` parameter, depending on chosen criteria.
+
+### Example Middleware
+
+Below is an example of basic authorization middleware, the full implementation of which comes with the server.
+
+```java
+public class AuthMiddleware implements Middleware {
+    ...
+
+    public Controller use(Controller controller) {
+        return (request) -> {
+            if (isAuthorized(request)) {
+                return controller.createResponse(request);
+            } else {
+                return createResponse();
```

Comment:
Both branches of this `if` statement call a `createResponse` method. Is there a way that you can modify this example to indicate what is happening in regards to the domain of HTTP? (i.e. it would be nice if the method was named  something like `unauthorized` instead of `createResponse`)